graders/c/cgrader/cgrader.py

```diff
@@ -370,7 +370,7 @@ def test_run(
         command,
         input=None,
         exp_output=None,
-        must_match_all_outputs=False,
+        must_match_all_outputs=False,  # True, False or 'partial'
```

Comment:
I realize this might not be totally backward compatible, but using an enum rather than `True`, `False`, and `"partial"` feels more user-friendly. At the very least, the code should throw an error message if the value isn't one of the specified ones.

Reply:
Throwing an error here may be backwards incompatible. I feel hesitant to introduce controls that didn't exist before.

Reply:
Replaced with enum in 13b15833d8a61efde6e9c24863f21e1a9053000a. @eliotwrobson it's my first time using typing in Python, can you check if it makes sense? I don't want to type-annotate the whole file in this PR, so this is just for this particular option.

Reply:
Sorry for the late reply on this. The enum is not necessary (in fact, `OutputMatchingOption.all == "all"` is False), but if you just want this for an annotation, you can do something simpler. See my latest comment.